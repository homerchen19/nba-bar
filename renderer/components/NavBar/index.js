import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { NavBar as AntdNavBar, Icon } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '@styles/theme';

const Wrapper = styled.div`
  width: 100vw;
  height: 30px;
`;

const Title = styled.p`
  color: ${colors.white};
  font-size: 15px;
`;

const NavBar = ({ page }) => (
  <Wrapper>
    <AntdNavBar
      leftContent={
        <Link href="/home">
          <Icon
            type="left"
            color={colors.white}
            style={{
              position: 'absolute',
              left: '7px',
              height: '30px',
              cursor: 'pointer',
            }}
          />
        </Link>
      }
      style={{ height: '100%', background: colors.darkBlue }}
    >
      <Title>{page}</Title>
    </AntdNavBar>
  </Wrapper>
);

NavBar.propTypes = {
  page: PropTypes.string.isRequired,
};

export default React.memo(NavBar);
