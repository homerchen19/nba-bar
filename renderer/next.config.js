const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
