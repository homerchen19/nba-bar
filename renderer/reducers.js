import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
