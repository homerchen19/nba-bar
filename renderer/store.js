import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
