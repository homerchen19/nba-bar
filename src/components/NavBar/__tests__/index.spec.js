import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import NavBar from '..';

it('be defined', () => {
  expect(NavBar).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    page: 'PAGE',
  };
  const tree = renderer
    .create(
      <MemoryRouter>
        <NavBar {...props} />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
