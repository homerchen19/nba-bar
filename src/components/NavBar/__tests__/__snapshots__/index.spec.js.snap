// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="NavBar__Wrapper-sc-1en1yvc-0 dZzGVK"
>
  <div
    className="am-navbar am-navbar-dark"
    style={
      Object {
        "background": "#084887",
        "height": "100%",
      }
    }
  >
    <div
      className="am-navbar-left"
      onClick={[Function]}
      role="button"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <svg
          className="am-icon am-icon-left am-icon-md"
          color="#f7f7f7"
          style={
            Object {
              "cursor": "pointer",
              "height": "30px",
              "left": "7px",
              "position": "absolute",
            }
          }
        >
          <use
            xlinkHref="#left"
          />
        </svg>
      </a>
    </div>
    <div
      className="am-navbar-title"
    >
      <p
        className="NavBar__Title-sc-1en1yvc-1 hnMWnd"
      >
        PAGE
      </p>
    </div>
    <div
      className="am-navbar-right"
    />
  </div>
</div>
`;
